### PR TITLE
backport: bitcoin 20564, 20661 (addrv2 follow-ups)

### PR DESCRIPTION
**Pull request by UdjinM6**
NOTE: fbacffe035ab51cdce82fa0c1a814ca1e28ba25d would make it incompatible with previous v0.18.x/v18.0.x versions (would disconnect peers), so I added a bit of backwards compatibility in 8c25b4c16c.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2608,6 +2608,15 @@ bool ProcessMessage(CNode* pfrom, const std::string& msg_type, CDataStream& vRec
...
         const CNetMsgMaker msg_maker(INIT_PROTO_VERSION);
+        // Signal ADDRv2 support (BIP155).
+        if (nSendVersion >= 70016) {
+            // BIP155 defines addrv2 and sendaddrv2 for all protocol versions, but some
+            // implementations reject messages they don't know. As a courtesy, don't send
+            // it to nodes with a version before 70016, as no software is known to support
+            // BIP155 that doesn't announce at least that protocol version number.
+            connman->PushMessage(pfrom, msg_maker.Make(NetMsgType::SENDADDRV2));
+        }
+
```
Comment (PastaPastaPasta) [inline src/net_processing.cpp:2618]: isn't this before mnauth?

File: test/functional/test_framework/messages.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 import dash_hash
 
 MIN_VERSION_SUPPORTED = 60001
-MY_VERSION = 70220  # ISDLOCK_PROTO_VERSION
+MY_VERSION = 70220 # ISDLOCK_PROTO_VERSION
```
Comment (PastaPastaPasta) [inline test/functional/test_framework/messages.py]: unrelated?

File: src/net_processing.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/net_processing.cpp:2618]: yes, but it's before VERACK so it's fine

Comment (PastaPastaPasta) [reply src/net_processing.cpp:2618]: so was the comment here wrong? https://github.com/dashpay/dash/pull/4874/files/fbacffe035ab51cdce82fa0c1a814ca1e28ba25d#diff-6875de769e90cec84d2e8a9c1b962cdbcda44d870d42e4215827e599e11e90e3L2717-L2720

L2717-2720

Comment (PastaPastaPasta) [reply src/net_processing.cpp:2618]: oh, in that location it was after verack, so a problem, but here it's before both verack and mnauth.

**Review by PastaPastaPasta — DISMISSED** [review]
okay, that makes sense then

~utACK for merging via merge commit~

Getting this error
```
net_processing.cpp:2754:20: error: 'nSendVersion' is a private member of 'CNode'
        if (pfrom->nSendVersion < ADDRV2_PROTO_VERSION) {
                   ^
./net.h:1095:9: note: declared private here
    int nSendVersion {0};
        ^
1 error generated.
```

**Comment by UdjinM6** [comment]
🙈 should be fixed now, pls test

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit; compiles now
